gusto/model.py

```diff
@@ -0,0 +1,136 @@
+from gusto.advection import SSPRK3, ThetaMethod
+from gusto.configuration import CompressibleParameters
+from gusto.forcing import ShallowWaterForcing, CompressibleForcing
+from gusto.linear_solvers import ShallowWaterSolver, CompressibleSolver
+from gusto.transport_equation import VectorInvariant, AdvectionEquation, SUPGAdvection, EulerPoincare
+
+
+class Model(object):
+
+    def __init__(self,
+                 state,
+                 physical_domain,
+                 parameters,
+                 timestepping,
+                 linear_solver,
+                 forcing,
+                 advected_fields,
+                 diffused_fields,
+                 physics_list):
+
+        self.state = state
+        self.physical_domain = physical_domain
+        self.parameters = parameters
+        self.timestepping = timestepping
+        self.linear_solver = linear_solver
+        self.forcing = forcing
+        self.advected_fields = advected_fields
+        self.diffused_fields = diffused_fields
+        self.physics_list = physics_list
+
+
+def ShallowWaterModel(state,
+                      physical_domain, *,
+                      parameters=None,
+                      timestepping=None,
+                      linear_solver=None,
+                      forcing=None,
+                      advected_fields=None,
+                      diffused_fields=None,
+                      physics_list=None):
+
+    if parameters is None:
+        raise ValueError("Default parameters cannot be set for shallow water model. You must at least provide the mean depth via the ShallowWaterParameters configuration class.")
+
+    if timestepping is None:
+        raise ValueError("Default timestepping parameters cannot be set. You must at least provide the timestep, dt, via the TimesteppingParameters configuration class.")
```

Comment:
If both of these things are required, best to make them required arguments, rather than optional.